language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/K8sCompletionRouter.java

```diff
@@ -0,0 +1,242 @@
+        addRooTablesToCompletions(snippets.keySet(), completions);
+
+        //Remove Existing document level entries from completion
+        List<String> existingChildEntries = findExistingChildEntries(arrayNode, completions);
+        return removeExistingChildTables(completions, existingChildEntries);
+    }
+
+    /**
+     * Find the node based on the cursor position.
+     */
+    public void fillNodeAtCursor(TomlCompletionContext context) {
+        Path tomlFilePath = context.filePath();
+        SyntaxTree st = TomlSyntaxTreeUtil.getTomlSyntaxTree(tomlFilePath).orElseThrow();
+        Position position = context.getCursorPosition();
+        TextDocument textDocument = st.textDocument();
+        int txtPos = textDocument.textPositionFrom(LinePosition.from(position.getLine(), position.getCharacter()));
+        context.setCursorPositionInTree(txtPos);
+        TextRange range = TextRange.from(txtPos, 0);
+        NonTerminalNode nonTerminalNode = ((DocumentNode) st.rootNode()).findNode(range);
+        while (nonTerminalNode.parent() != null && !TomlSyntaxTreeUtil.withinTextRange(txtPos, nonTerminalNode)) {
+            //Takes the parent Node of the immediate cursor position
+            nonTerminalNode = nonTerminalNode.parent();
+        }
+        context.setNodeAtCursor(nonTerminalNode);
+    }
+
+    private Map<String, CompletionItem> getTableCompletions(TableNode tableNode,
+                                                            Map<Parent, Map<String, CompletionItem>> snippets) {
+
+        String tableKey = TomlSyntaxTreeUtil.toDottedString(tableNode.identifier());
+        Map<String, CompletionItem> completions = new HashMap<>(getChildFromDottedKey(snippets, tableKey));
+
+        for (KeyValueNode field : tableNode.fields()) {
+            String key = TomlSyntaxTreeUtil.toDottedString(field.identifier());
+            completions.remove(key);
+        }
+        //Add main completions to completions
+        addRooTablesToCompletions(snippets.keySet(), completions);
+
+        //Remove Existing document level entries from completion
+        List<String> existingChildEntries = findExistingChildEntries(tableNode, completions);
+        return removeExistingChildTables(completions, existingChildEntries);
+    }
+
+    private Map<String, CompletionItem> addRooTablesToCompletions(Set<Parent> parents,
+                                                                  Map<String, CompletionItem> completions) {
+        for (Parent parent : parents) {
+            String key = parent.getKey();
+            if (completions.containsKey(key)) {
+                continue;
+            }
+            ParentType type = parent.getType();
+            CompletionItem item = new CompletionItem();
+            if (type == ParentType.TABLE) {
+                Table objectNode = new Table(key);
+                item.setInsertText(objectNode.prettyPrint());
+                item.setDetail(TomlSyntaxTreeUtil.TABLE);
```

Comment:
Here are some common field settings duplicated. Maybe we can have a util method to do this